python/interpret_community/widget/explanation_dashboard.py

```diff
@@ -55,18 +82,66 @@ class and for the regression case a method of predict() returning the prediction
     default_template = env.get_template("inlineDashboard.html")
 
     class DashboardService:
-        app = Flask(__name__)
-        CORS(app)
 
-        def __init__(self, port):
+        def __init__(self, port, public_ip, with_credentials=False):
+            app = Flask(__name__)
+            self.nbvm = _get_nbvm()
+            if self.nbvm is None and not with_credentials:
+                self.cors = CORS(app)
+                self.with_credentials = False
+                self.ip = LOCALHOST
+                self.env = 'local'
+            elif self.nbvm is not None:
+                logging.getLogger('flask_cors').level = logging.DEBUG
+                instance_name = self.nbvm["instance"]
+                domain_suffix = self.nbvm["domainsuffix"]
+                nbvm_origin1 = "https://{}.{}".format(instance_name, domain_suffix)
+                nbvm_origin2 = "https://{}-{}.{}".format(instance_name, port, domain_suffix)
+                nbvm_origins = [nbvm_origin1, nbvm_origin2]
+                headers = ['Content-Type']
+                # Support credentials for notebook VM scenario
+                self.cors = CORS(app, origins=nbvm_origins, expose_headers=headers, supports_credentials=True)
+                self.with_credentials = True
+                self.ip = LOCALHOST
+                self.env = 'azure'
+            else:
+                if public_ip is not None:
```

Comment:
The use case of azure NB VM seems to not require with_credentials or public_ip be provided, and if they were provided, its not clear that everything would work as expected?

Reply:
Azure NBVM requires with_credentials.  Local case doesn't require either parameters.  DSVM requires public ip address.  Some possible (unknown) VM like DSVM but with credentials may been both public_ip and with_credentials - I tried to setup just this (DSVM with credentials middle-layer) but ran into other issues.